static/js/copyPasteEvents.js

```diff
@@ -130,12 +242,84 @@ var htmlDecode = function(input) {
+// P.S: It's not possible to have two or more comments when there is only text selected, because for each comment
+// created it's generated a <span> and to copy only the text it MUST NOT HAVE any tag on the selection
+exports.getCommentIdOnFirstPositionSelected = function() {
+  var attributeManager = this.documentAttributeManager;
+  var rep = this.rep;
+  var commentId = _.object(attributeManager.getAttributesOnPosition(rep.selStart[0], rep.selStart[1])).comment;
+  return commentId;
+};
+
+exports.hasCommentOnSelection = function() {
+  var hasComment;
   var attributeManager = this.documentAttributeManager;
   var rep = this.rep;
-  var selStartAttrib = _.object(attributeManager.getAttributesOnPosition(rep.selStart[0], rep.selStart[1])).comment;
-  var selEndAttrib = _.object(attributeManager.getAttributesOnPosition(rep.selEnd[0], rep.selEnd[1] - 1)).comment;
-  return selStartAttrib === selEndAttrib ? selStartAttrib : null;
+  var firstLineOfSelection = rep.selStart[0];
+  var firstColumn = rep.selStart[1];
+  var lastColumn = rep.selEnd[1];
+  var lastLineOfSelection = rep.selEnd[0];
+  var selectionOfMultipleLine = hasMultipleLineSelected(firstLineOfSelection, lastLineOfSelection);
+
+  if(selectionOfMultipleLine){
+    hasComment = hasCommentOnMultipleLineSelection(firstLineOfSelection,lastLineOfSelection, rep, attributeManager);
+  }else{
+    hasComment = hasCommentOnLine(firstLineOfSelection, firstColumn, lastColumn, attributeManager)
+  }
+  return hasComment;
+};
+
+var hasCommentOnMultipleLineSelection = function(firstLineOfSelection, lastLineOfSelection, rep, attributeManager){
+  var line = firstLineOfSelection;
+  var multipleLineSelectionHasComment = false;
+  for (line; line <= lastLineOfSelection; line++) {
+    var firstColumn = getFirstColumnOfSelection(line, rep, firstLineOfSelection);
+    var lastColumn = getLastColumnOfSelection(line, rep, lastLineOfSelection);
+    var hasComment = hasCommentOnLine(line, firstColumn, lastColumn, attributeManager)
+    if (hasComment){
+      multipleLineSelectionHasComment = true;
+    }
+  }
+  return multipleLineSelectionHasComment;
+};
+
+var getFirstColumnOfSelection = function(line, rep, firstLineOfSelection){
+  return line !== firstLineOfSelection ? 0 : rep.selStart[1];
+};
+
+var getLastColumnOfSelection = function(line, rep, lastLineOfSelection){
+  var lineLength = getLength(line, rep);
+  var positionOfLastCharacterSelected = rep.selEnd[1] - 1;
+  return line !== lastLineOfSelection ? lineLength : positionOfLastCharacterSelected;
+};
+
+var hasCommentOnLine = function(lineNumber, firstColumn, lastColumn, attributeManager){
+  var column = firstColumn;
+  var hasComment = false;
+  for (column; column <= lastColumn; column++) {
+   var commentId = _.object(attributeManager.getAttributesOnPosition(lineNumber, column)).comment;
+   if (commentId !== undefined){
+    hasComment = true;
+   }
+  }
+  return hasComment;
```

Comment:
same here as on `hasCommentOnMultipleLineSelection()`: you don't need to keep checking next chars on the line if you've already found one char with comment. Use the `hasComment` flag to break the `for`